Bottom: 4fcfc21b6cfe98f1975a9474b05cf717402a287b
Top:    37a3483e543f3628a4aa3e50b59ed4718d3b90a9
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-24 20:16:38 +0000

polygon.c: Temporarily disable coalescing adjacent line segments

We need to extend the tests to properly account for arc segments.


---

diff --git a/src/polygon.c b/src/polygon.c
index c2da6eb..fa616ad 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1364,6 +1364,7 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
   if (Undoing ())
     return (false);
 
+if (0) {
   for (n = 0; n < Polygon->PointN; n++)
     {
       prev = prev_contour_point (Polygon, n);
@@ -1373,6 +1374,7 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
       line.Point1 = Polygon->Points[prev];
       line.Point2 = Polygon->Points[next];
       line.Thickness = 0;
+#warning THIS TEST IS NOT SUFFICIENT.. DOESNT ACCOUNT FOR DIFFERENT ARC CENTERS
       if (Polygon->Points[prev].included_angle == Polygon->Points[n].included_angle &&
           IsPointOnLine (p->X, p->Y, 0.0, &line))
         {
@@ -1380,6 +1382,7 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
           changed = true;
         }
     }
+}
   return (changed);
 }
